consensus/bor/heimdallgrpc/span.go

```diff
@@ -0,0 +1,58 @@
+package heimdallgrpc
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/consensus/bor/heimdall/span"
+	"github.com/ethereum/go-ethereum/consensus/bor/valset"
+
+	proto "github.com/maticnetwork/polyproto/heimdall"
+	protoutils "github.com/maticnetwork/polyproto/utils"
+)
+
+func (h *HeimdallGRPCClient) Span(ctx context.Context, spanID uint64) (*span.HeimdallSpan, error) {
+	req := &proto.SpanRequest{
+		ID: spanID,
+	}
+
+	res, err := h.client.Span(context.Background(), req)
+	if err != nil {
+		return nil, err
```

Comment:
Won't we need a retry mechanism, specifically for `span`? Also, I think it's good to have it also for state-sync & checkpoint methods.

Reply:
Why you use blank context here?

Reply:
Fixed the context

Reply:
Good point regarding retry, I will check

Reply:
Fixed the context, Looking to implement retry mechanism.

Reply:
Implemented retry middleware, Now need to add supported errors on heimdall side.